doc/develop/tools/clion.rst

```diff
@@ -0,0 +1,270 @@
+.. _clion_ide:
+
+..
+   Copyright 2023 Marina Kalashina <marina.kalashina@jetbrains.com>
+
+
+CLion
+#####
+
+`CLion <https://www.jetbrains.com/clion/>`_ is a cross-platform C/C++ IDE that
+supports multi-threaded RTOS debugging.
+
+This guide describes the process of setting up, building, and debugging
+Zephyr's :ref:`hello_world` sample in CLion.
+
+Get CLion
+*********
+
+Download_ and install CLion.
+
+.. _Download: https://www.jetbrains.com/clion/download
+
+Initialize a new workspace
+**************************
+
+This guide gives details on how to build and debug the
+:ref:`hello_world` sample application, but the instructions would
+be similar for any Zephyr project and
+:ref:`workspace layout west-workspaces`.
+
+Before you start, make sure you have a working Zephyr development
+environment, as per the instructions in the :ref:`getting_started`.
+
+Open the project in CLion
+**************************
+
+#. In CLion, click **Open** on the Welcome screen or select **File | Open**
+   from the main menu.
+
+#. Locate your Zephyr workspace folder (``~/zephyrproject`` if you
+    followed the Getting Started instructions) and select it.
+
+   Click **OK**.
+
+#. If prompted, click **Trust Project**.
+
+   See `this section`_ in CLion web help for more information on project
+   security.
+
+   .. _this section: https://www.jetbrains.com/help/clion/project-security.html#projects_security
+    </chapter>
+
+#. If the toolchain is not yet configured, CLion will open the
```

Comment:
wouldn't we want to have folks use the Zephyr toolchain instead? or maybe it doesn't matter as west/ninja then take over anyway?

Reply:
Might be it's better to add that the compilers do not matter, zephyr uses it's own SDK anyway